File: python/tokens/tokens.py
#!/usr/bin/python3 
# token objects that store their values, properties, privileges, etc. 

import os 
import sys

class Token(): 
    def __init__ (self): 
        pass

class Number(Token):
    def __init__ (self, value): 
        self.value = value 
    
    def __str__ (self): 
        return str(self.value)

class Variable(Token): 
    def __init__ (self, value): 
        self.value = value 
    
    def __str__ (self): 
        return self.value 

class Brackets(Token): 
    def __init__ (self, start, content, end): 
        self.start = start, 
        self.content = content,  
        self.end = end 
        self.privilege = 5 
    
    def __str__ (self): 
        return self.start + str(self.content) + self.end

class Operator(Token): 
    def __init__ (self, left, right, privilege, label): 
        self.left = left 
        self.right = right 
        self.privilege = privilege
        self.label = label 
    
    def __str__ (self, level=0): 
        return str(self.left) + self.label + str(self.right)

File: react-app/src/resources/tokens.js
export class Token { }

export class NumberToken extends Token {
    constructor (value) {
        super(); 
        this.value = value; 
    }
}

export class VariableToken extends Token {
    constructor (value) {
        super(); 
        this.value = value; 
    }
}

export class PointerToken extends Token {
    constructor (point) {
        super(); 
        this.point = point; 
    }

    fetch_destination () {
        return this.point; 
    }
}

export class BracketsToken extends Token {
    constructor (start, content, end) {
        super(); 
        this.start = start; 
        this.content = content; 
        this.end = end; 
        this.privilege = 5; 
    }
}

export class OperatorToken extends Token {
    constructor (left, right, privilege, label) {
        super(); 
        this.left = left; 
        this.right = right; 
        
        this.privilege = privilege; 
        this.label = label; 
    }
}